documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
 Each section addresses a dimension of Swagger authoring and its effectiveness for delivering the best Azure developer experiences through tools and libraries built using a Swagger specification.
 
 ## Notational Conventions ##
-The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](http://www.ietf.org/rfc/rfc2119.txt).
+
+The keywords "*MUST*", "*MUST NOT*", "REQUIRED", "SHALL", "SHALL NOT", "*SHOULD*", "*SHOULD NOT*", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](http://www.ietf.org/rfc/rfc2119.txt).
 
 Normative statements of the checklist are presented in the following manner:
 
-&#x25a2; *Mnnnn*: checklist text here.
+- [ ] **Mnnnn**: checklist text here.
 
 OR
 
-&#x25a2; *Snnnn*: checklist text here.
+- [ ] **Snnnn**: checklist text here.
 
-Where "M" stands for 'MUST' or 'MUST NOT' requirement. "S" stands for 'SHOULD' or 'SHOULD NOT' requirement. And, "nnnn" is replaced by a number that is unique among all the requirements in the checklist, thereby "nnnn" becomes a unique requirement identifier.
+Where "M" stands for '*MUST*' or '*MUST NOT*' requirement. "S" stands for '*SHOULD*' or '*SHOULD NOT*' requirement. And, "nnnn" is replaced by a number that is unique among all the requirements in the checklist, thereby "nnnn" becomes a unique requirement identifier.
 
 
 ## Naming - Swagger Checklist ##
 
-&#x25a2; *M1000*: Plural names MUST only be used to represent collections.
+- [ ] **M1000**: Plural names *MUST* only be used to represent collections.
+
+- [ ] **M1001**: Each operation must contain an OperationId of the form "RESOURCE_METHOD" a.k.a "NOUN_VERB", where "RESOURCE" represents the resource or object the operation addresses, and METHOD is a short name for the operation being performed "Get, Start, Delete, Create, List", etc.
+
+- [ ] **M1002**: If the NOUN in the OperationId "NOUN_VERB" matches with the name of the Model definition then the NOUN in the operationId *MUST* use the plural form of NOUN to avoid collision in the namespace.
+
+- [ ] **M1003**: List operations *MUST* be prefixed with "List" and *MUST NOT* be named as "GetXXXX" operations.
+
+- [ ] **M1004**: List operations *MUST* use consistent names and semantics. List operations *MUST* NOT use any other names. Consistent names and semantics are:
+
+  - `List()` - lists all resources under a subscription.
+  - `ListByResourceGroup()` - list all resources in a resource group within a subscription.
```

Comment:
This is primarily applicable to ARM Resource Providers. We need to be clear on which rules apply under what circumstances (i.e. a data plane swagger would typically not adhere to this guidance).